### PR TITLE
Make backend methods async

### PR DESCRIPTION
**Pull request by martinvonz**
# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

**Comment by martinvonz** [comment]
Seems harmless but I'll wait until after 0.22 to merge this just in case

File: lib/src/store.rs [hunk 1]
```diff
@@ -262,7 +262,7 @@ impl Store {
         self.backend.read_symlink(path, id).await
     }
 
-    pub fn write_symlink(&self, path: &RepoPath, contents: &str) -> BackendResult<SymlinkId> {
+    pub async fn write_symlink(&self, path: &RepoPath, contents: &str) -> BackendResult<SymlinkId> {
         self.backend.write_symlink(path, contents).block_on()
```
Comment (yuja) [inline lib/src/store.rs:266]: `.await`

**Review by yuja — APPROVED** [review]
Do we plan to remove sync version of `store.read`/`get_<thing>()` and drop `_async` suffix? The current function naming is inconsistent.

File: lib/src/store.rs (hunk 1 above)
Comment (martinvonz) [reply lib/src/store.rs]: Done.

**Comment by martinvonz** [comment]
> Do we plan to remove sync version of `store.read`/`get_<thing>()` and drop `_async` suffix? The current function naming is inconsistent.

Yes, I think we should. I also noticed that as I was working on this PR.

**Comment by PhilipMetzger** [comment]
I think it's worthwhile to convert the `get_commit()` to return a future and instead add a `get_commit_sync()` helper for those who still need a sync method. The second method is only needed, if we have enough callers for it.

**Pull request merged**